test/OpenTelemetry.Instrumentation.EntityFrameworkCore.Tests/EntityFrameworkDiagnosticListenerTests.cs

```diff
@@ -70,6 +70,39 @@ public void EntityFrameworkContextEventsInstrumentedTest()
         VerifyActivityData(activity);
     }
 
+    [Fact]
+    public void EntityFrameworkContextWithAlternativeDisplayName()
+    {
+        var activityProcessor = new Mock<BaseProcessor<Activity>>();
+        var altDisplayName = "AltName";
+        using var shutdownSignal = Sdk.CreateTracerProviderBuilder()
+            .AddProcessor(activityProcessor.Object)
+            .AddEntityFrameworkCoreInstrumentation(options =>
+            {
+                options.EnrichWithIDbConnection = (activity1, _) =>
+                {
+                    activity1.DisplayName = altDisplayName;
```

Comment:
I think a better test would be one that actually makes use of the payload to set the Activity Display name or the tag.

Reply:
Something like this? (CommandText wouldn't be available at this point)
 
   var systemName = activity.GetTagItem("db.system");
   activity.DisplayName = systemName;

Reply:
Updated, it grabs the state and includes name now. :)